### DISCUSSION TITLE
Count down for results announcement?

### DISCUSSION THREAD

**Discussion by Jezzamonn** [ludumdare/ludumdare#2015] [General]
There's a lot of hype around the theme announcement, but when the results are revealed it seems like there's not as much excitement. So my idea is to have a specific time that the results are always announced. Something like exactly 21 days after the jam ends, the rating results are revealed. So that Mike has time to write up the closing post and finalise everything, the rating could end 20 days after the jam. Of course, those numbers could change too, but the idea is to have a specific known time when they're released. That way as the time is coming up people can be excited knowing that there's not long to find out how they went!

It'd probably result in another spike in site traffic, but it'd still be cool

**Comment by mikekasprzak** [comment]
I'm going to assume what you're really saying is that results should be more interesting, 'cause we've sort of always done what you're describing. Just in recent events I've chosen to be more vague with results time to give me time to make sure new systems were working correctly, and... well lets talk about that. :smile:

I will say that a certain video streaming platform **REALLY REALLY** wants us to do _Live Show_ of some sort, and in the back of my mind that has always been a _Results Show_. There's a number of steps that need to be taken to get there, and I think we're still at least a year away from this.

Ludum Dare starts when it does because I'm based in North America (Canada), and looking to maximize the amount of time people on the west coast can work on games without skipping work (i.e. the event starts at 6 PM Friday). Sadly this sucks for those in the EU (start times ranging from midnight to 3 AM).

_Results Show_ however, we can push that back a bit more to make it EU friendly. It's pretty common for us in NA to have a Nintendo Direct or E3 press conference in the middle of the afternoon. You either tune in, catch the highlights on social media, or watch a VOD later. Finding out that you won LD while you're at work or school isn't a bad thing. It can lead to an impromptu after party with friends and all. :wink:

My tentative time for the results show would be ~5 hours earlier than an LD start time (like 1 PM or 2 PM on the west coast). This is partially why I've started pushing results to Wednesday afternoon, just to start getting folks used to the new time. Obviously it's just a data drop for us right now (i.e. click results), but I would like to see this grow in to a bigger more extravagant presentation. I'm envisioning a 30 minute program highlighting some of the highest rated games, then soon after (or even during the broadcast) full results drop on the website

**Comment by local-minimum** [comment]
@povrazor I really like to see this happen and I know we've discussed it before, but: what's the roadmap, and what can be made to make us get there faster? Maybe it is time for projects here on github or what they are called and this could be one of them: Have a winner's show.

**Comment by mikekasprzak** [comment]
Sure. Here's some more thoughts.

## Event Structure Changes
Preparing for a live show takes time. This means that the end of the 3 week voting period needs to change a bit. Specifically, it means the normal Play and Rate stage ends early, and the final few days become something different.

**Semi-results**: Basically the top X games from all categories need to be calculated, and they get added to a public pool. So at this point people can know if they're a "Top Game", but they don't know exactly where they are placed and why.

Preparing a ~30 minute show, I'm going to need a bunch of help capturing 15-30 second clips of the "Top Games". Over time finding help is going to be tough (i.e. people excited to do it eventually dropping out), so it's important for longevity that anyone is able to just decide to spontaneously help out, and this is what I've come up with as a way to crowd-source the largest chunk of the work.

Of course, there needs to be a system in place to upload and catalog these video, and a way for me/someone to approve a clip, so others know it's good.

Clips should be naked, pure gameplay footage, with very minor editing in case interesting moments should be highlighted. 

Here's a couple examples, though in the case of the presentation it would be purely gameplay footage, not timelapse.

https://www.youtube.com/watch?v=S6N81Z8nfC8
https://www.youtube.com/watch?v=gyVOeXgYr44
https://www.youtube.com/watch?v=avwiyQgn1Ew

In essence we'd be building a system for creating these sort of videos. In fact I would love to re-purpose this to actually create more of these videos. I really loved having these videos leading up to the event.

They key difference though is those were made by hand, and what we're shooting for is an automated version, using a variety of tools (FFMpeg, PHP image libraries, ImageMagick if needed, etc).

## Video Processing
Clips get uploaded, then a server does some processing on the videos, preparing them for the show. Like the above videos, overlays would have to be generated, and faded/transitioned in accordingly.

* Game Name
* Author name(s)
* What awards were won

I should then be able to download all the clips, drag them in to Sony Vegas, record a few small voice-overs (hello, welcome to the results show for LD4x), insert a few bumpers, and export. So basically it's not actually a fully live show, but more Nintendo Direct like. At least until it's practical to do something with a studio setup. No matter which way we go though (edited vs studio live), we're going to need game clips to play, 'cause live just isn't practical (maybe for an after show though).

There will also need to be some housekeeping items, like a credits roll generator, where the crew and everyone that helped edit clips gets a nod. Also a pre-roll generator, basically a countdown until the stream starts. A lot like LD itself, the more we can automate, the easier it is to run.

I'm sure I've missed some details, but I think that's a good place to start this discussion.

**Comment by mikekasprzak** [comment]
Okay, a more specific breakdown.

* Generate a list of "Top Games" list
* Provide an upload guide (resolution, file format, framerate, compressor)
* Allow users to upload video clips of games
* Allow users to recommend changes to metadata, such as twitter accounts for the team
* Allow me/staff to approve an upload, or reject and leave a reason
* Do things to help users from stomping over each-others work. i.e. let people say they're working on a clip, but have a timeout, in case someone signs up but doesn't have time to do the work. You should only be signing up if you are actually making the clip at that moment, 'cause there is a lot of work to be done and not many days to do it (2-3).
* Generate overlays/lower thirds for a clip (actual results are kept secret, for my eyes only)
* Render videos with overlays to correct format (MP4 probably)
* Make it easy for me to download everything
* Tell me the order they should go in (reverse order, no repeats, all notable scores get mentioned in the overlays)

This is the bulk of the work.

For the sake of completeness, some other jobs that need doing:

* Generate a credits roll, everybody that worked on the production, and helped edit videos
* Generate pre-roll, i.e. what is shown before the stream starts.
* Make some bumpers.
  * Stuff for the beginning/intro. a short brief on what you're about to watch.
  * Bumpers per section
    * a bunch of games from the compo/jam that didn't win, but still scored very highly. It's notable that the games that win a Ludum Dare don't necessarily get turned in to fancy commercial games, so the more we can show the better.
    * bumpers for the categories (if they didn't also win overall... or we specifically leave that a `?`)
    * bumpers for the top 5 overall in the compo, and then the jam.
  * wrapup, thanks for watching, then roll credits.

Eventually once user categories are in and mature, some of the better ones can be included to again highlight more interesting games.

The goal is a ~30 minute presentation, somewhat similar to a Nintendo Direct (prepared rather than live, so I have the luxury of not being at home, in case I'm somewhere with terrible internet access). In time it could evolve in to a live presentation in a studio setup, but live makes result times less predictable. So alternatively it may be better that this part stays prepared, and the studio thing becomes a post-show where whomever is there chats about the results, and plays a few games (where appropriate).
